app/src/main/java/net/bible/android/control/backup/BackupControl.kt

```diff
@@ -93,6 +95,39 @@ class BackupControl @Inject constructor() {
         }
     }
 
+    /** Backup database to Uri returned from ACTION_CREATE_DOCUMENT intent
+     */
+    fun backupDatabaseToUri( uri: Uri) : Boolean {
+        val out = BibleApplication.application.contentResolver.openOutputStream(uri)!!
+        val filename = DATABASE_NAME;
+        val f = File(internalDbDir, filename);
+        val inputStream = FileInputStream(f)
+        val header = ByteArray(16)
+        inputStream.read(header)
+        var ok = false
+        if (String(header) == "SQLite format 3\u0000") {
```

Comment:
Why are we checking format of a known db file we are writing? Checking is needed when restoring db, but not so much when writing, I think.